harness/determined/common/experimental/resource_pool.py

```diff
@@ -0,0 +1,113 @@
+from typing import List, Optional
+
+from determined.common import api
+from determined.common.api import bindings
+from determined.common.experimental import workspace
+
+
+class ResourcePool:
+    """A class representing a resource pool object.
+
+    Attributes:
+        name: (str) The name of the resource pool.
+    """
+
+    def __init__(
+        self,
+        session: api.Session,
+        name: Optional[str] = None,
+    ):
+        """Create a resource pool object.
+
+        Arguments:
+            session: The session to use for API calls.
+            workspace_name: (Optional) Name of the resource pool.
+        """
+        self.name = name
+        self._session = session
+
+    @classmethod
+    def _from_bindings(
+        cls, resource_pool_bindings: bindings.v1ResourcePool, session: api.Session
+    ) -> "ResourcePool":
+        resource_pool = cls(session, name=resource_pool_bindings.name)
+        return resource_pool
+
+    def bind(
+        self,
+        workspace_names: List[str],
+    ) -> None:
+        """
+        Bind a resource pool to workspaces.
+
+        Arguments:
+            workspace_names (list(str)): The names of the workspaces to be bound.
+        """
+        req = bindings.v1BindRPToWorkspaceRequest(
+            resourcePoolName=self.name,
+            workspaceNames=workspace_names,
+        )
+
+        bindings.post_BindRPToWorkspace(self._session, body=req, resourcePoolName=self.name)
```

Comment:
I haven't spent as much time with `bindings` as you'd expect, but it seems to me that some functions take just straight parameters, and some take a constructed request. The former seems more user-friendly (and I think is more common? Am unsure).

What's the motivation behind first creating a request and then sending that request, here?

side note: whatever the answer, it's a little weird that the `resourcePoolName` is needed in both.

Reply:
yeah this is the first time I'm working with bindings too, so I'm not sure why some are requests and others are parameters. It seems like requests are used when there are more than one parameter to pass? The `resourcePoolName` parameter here could be translated from the proto, which has `resourcePoolName` as part of the URL and as part of the request

Reply:
I'm also new to this. Just to share my findings in case people also want to know. It's due to how we define proto.

Payload definition:
```
message BindRPToWorkspaceRequest {
  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
    json_schema: { required: [ "resource_pool_name" ] }
  };

  // The resource pool name.
  string resource_pool_name = 1;
  // The workspace IDs to be bound to the resource pool.
  repeated int32 workspace_ids = 2;
  // The workspace names to be bound to the resource pool.
  repeated string workspace_names = 3;
}
```
API endpoint definition:
```
  rpc BindRPToWorkspace(BindRPToWorkspaceRequest)
      returns (BindRPToWorkspaceResponse) {
    option (google.api.http) = {
      post: "/api/v1/resource-pools/{resource_pool_name}/workspace-bindings",
      body: "*"
    };
    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
      tags: "Internal"
    };
  }
```

Reply:
Yeah I'm not sure this is worth investigating, but perhaps something to look into for a tech debt ticket? Does seem interesting